numba/compiler.py

```diff
@@ -179,6 +142,22 @@ def compile_isolated(func, args, return_type=None, flags=DEFAULT_FLAGS,
                              flags, locals)
 
 
+def run_frontend(func):
+    """
+    Run the compiler frontend over the given Python function, and return
+    the function's canonical Numba IR.
+    """
+    # XXX make this a dedicated Pipeline?
```

Comment:
As the comment suggests, I think we should make it a separate pipeline (i.e. split the current unified pipeline in three sub-pipelines). Do you think that makes sense?